### PR TITLE
upgrade mypy to 0.971

### PR DESCRIPTION
**Pull request by gruebel**
It supersedes #582

I upgraded `mypy` to the latest version and added the missing stub libs, which are now separately distributed and packaged within `typeshed` anymore.

### CODE REVIEW

**Comment by gruebel** [comment]
@jpdakran would be great to get an opinion on it 🙂

File: requirements-dev.txt [hunk 1]
```diff
@@ -35,6 +35,8 @@ toml==0.10.2
 tox==3.24.4
 tox-pip-extensions==1.6.0
 typed-ast==1.4.3
+types-PyYAML==6.0.0
```
Comment (jpdakran) [inline requirements-dev.txt:38]: Can we upgrade this to types-PyYAML== 6.0.11?

File: requirements-dev.txt [hunk 2]
```diff
@@ -35,6 +35,8 @@ toml==0.10.2
...
+types-requests==2.26.0
```
Comment (jpdakran) [inline requirements-dev.txt:39]: And this to 2.28.9?

**Comment by gruebel** [comment]
@jpdakran would be great to get a review for this again 🙂

**Comment by gruebel** [comment]
@lorenzodb1 or @jpdakran maybe interested in reviewing this 🥺

**Review by jpdakran — APPROVED** [review]

File: detect_secrets/transformers/yaml.py [hunk 3]
```diff
@@ -210,25 +210,28 @@ def _compose_node_shim(
...
             else self.loader.line
         )
 
-        node = yaml.composer.Composer.compose_node(self.loader, parent, index)
-        node.__line__ = line + 1
+        node = yaml.composer.Composer.compose_node(self.loader, parent, index)  # type: ignore
+        if node is None:
+            return None
+
+        node.__line__ = line + 1    # type: ignore
```
Comment (lorenzodb1) [inline detect_secrets/transformers/yaml.py:224]: Could you please expand on these changes? I'm unsure on what you're trying to achieve here.

Comment (gruebel) [reply detect_secrets/transformers/yaml.py:224]: basically the yaml stubs suggest a return value of either `Node | None`, therefore I wanted to make sure it is not `None`.

**Review by lorenzodb1 — APPROVED** [review]

**Comment by lorenzodb1** [comment]
Thank you for your contribution @gruebel!

**Pull request merged**